Tests/Integration/MSFT_XIISLogging.Integration.Tests.ps1

```diff
@@ -0,0 +1,95 @@
+            } | Should not throw
+        }
+
+        It 'should be able to call Get-DscConfiguration without throwing' {
+            { Get-DscConfiguration -Verbose -ErrorAction Stop } | Should Not throw
+        }
+        #endregion
+
+        It 'Changing Loggging Rollover Settings ' -test {
+
+            Invoke-Expression -Command "$($Global:DSCResourceName)_Rollover -OutputPath `$TestEnvironment.WorkingFolder"
+            Start-DscConfiguration -Path $TestEnvironment.WorkingFolder -ComputerName localhost -Wait -Verbose -Force
+            
+            $CurrentLogSettings = Get-WebConfiguration -filter '/system.applicationHost/sites/siteDefaults/Logfile'
+               
+            $CurrentLogSettings.directory | Should Be 'C:\IISLogFiles'
+            $CurrentLogSettings.logExtFileFlags | Should Be 'Date,Time,ClientIP,UserName,ServerIP'
+            $CurrentLogSettings.logformat | Should Be 'W3C'
+            $CurrentLogSettings.period | Should Be 'Hourly'
+            $CurrentLogSettings.localTimeRollover | Should Be 'True'
+            $CurrentLogSettings.logformat | Should be 'W3C'
+        }
+    }    
+    
+    Describe "$($Global:DSCResourceName)_Truncate" {
+        #region DEFAULT TESTS
+        It 'Should compile without throwing' {
+            {
+                Invoke-Expression -Command "$($Global:DSCResourceName)_Truncate -OutputPath `$TestEnvironment.WorkingFolder"
+                Start-DscConfiguration -Path $TestEnvironment.WorkingFolder -ComputerName localhost -Wait -Verbose -Force
+            } | Should not throw
+        }
+
+        It 'should be able to call Get-DscConfiguration without throwing' {
+            { Get-DscConfiguration -Verbose -ErrorAction Stop } | Should Not throw
+        }
+        #endregion
+        It 'Changing Loggging Truncate Settings ' -test {
+
+            Invoke-Expression -Command "$($Global:DSCResourceName)_Truncate -OutputPath `$TestEnvironment.WorkingFolder"
+            Start-DscConfiguration -Path $TestEnvironment.WorkingFolder -ComputerName localhost -Wait -Verbose -Force
+            
+            $CurrentLogSettings = Get-WebConfiguration -filter '/system.applicationHost/sites/siteDefaults/Logfile'
+               
+            $CurrentLogSettings.directory | Should Be 'C:\IISLogFiles'
+            $CurrentLogSettings.logExtFileFlags | Should Be 'Date,Time,ClientIP,UserName,ServerIP'
+            $CurrentLogSettings.logformat | Should Be 'W3C'
+            $CurrentLogSettings.TruncateSize | Should Be '2097152'
+            $CurrentLogSettings.localTimeRollover | Should Be 'True'
+            $CurrentLogSettings.logformat | Should be 'W3C'    
+        }
+    }
+}
+finally
+{
+    #region FOOTER
+    Restore-WebConfiguration -Name $tempName
+    Remove-WebConfigurationBackup -Name $tempName
+
+    Restore-TestEnvironment -TestEnvironment $TestEnvironment
+    #endregion
+}
```

Comment:
BTW- A fantastic job on the integration tests! Really good to see effective integration testing.